src/ashley/editor/decorators.py

```diff
@@ -59,3 +59,34 @@ def mention(props):
         )
 
     return None
+
+
+def image(props):
+    """
+    Decorator for the `IMAGE` entity in Draft.js ContentState. Image can be decorated with a width,
+    height, and alignment property. By default draft-js insert data at 40% width of the editor,
+    we want to keep the same ratio so that the preview is alike.
+    """
+    css_class = ""
+    alignment = props.get("alignment", None)
+    if alignment in ("left", "right"):
+        css_class = f"float-{alignment}"
+    if alignment == "center":
+        css_class = "image-center"
+
+    # make sure data are number and positive, default is 40% of editor size
+    width = f'{abs(int(props.get("width", "40")))}%'
+
+    height = "auto"
+    if props.get("height", None) is not None:
+        height = f'{abs(int(props.get("height")))}%'
```

Comment:
Why isn't it done with CSS?
This HTML be stored in database and it will be painful to update if you want to change these values later.

Reply:
what do you mean done in css ? height and width could be 1 to 100%, I don't see how to deal with this directly in css, could you give me a hint on that please :-)

Reply:
You are right, I had misunderstood this code. Sorry you can ignore my comment.